lib/phoenix_live_view/upload_channel.ex

```diff
@@ -0,0 +1,139 @@
+defmodule Phoenix.LiveView.UploadChannel do
+  @moduledoc false
+  use Phoenix.Channel, log_handle_in: false
+
+  require Logger
+
+  alias Phoenix.LiveView.{Static, Channel}
+
+  def cancel(pid) do
+    GenServer.call(pid, :cancel)
+  end
+
+  def consume(pid, entry, func) when is_function(func, 1) or is_function(func, 2) do
+    case GenServer.call(pid, {:consume, entry, func}) do
+      {:ok, result} ->
+        result
+
+      {:error, :in_progress} ->
+        raise RuntimeError, "cannot consume uploaded file that is still in progress"
+    end
+  end
+
+  @impl true
+  def join(_topic, auth_payload, socket) do
+    %{"token" => token} = auth_payload
+
+    with {:ok, %{pid: pid, ref: ref}} <- Static.verify_token(socket.endpoint, token),
+         {:ok, config} <- Channel.register_upload(pid, ref),
+         %{max_file_size: max_file_size, chunk_timeout: chunk_timeout} = config,
+         {:ok, path} <- Plug.Upload.random_file("live_view_upload"),
+         {:ok, handle} <- File.open(path, [:binary, :write]) do
+      Process.monitor(pid)
+
+      socket =
+        assign(socket, %{
+          path: path,
+          handle: handle,
+          live_view_pid: pid,
+          max_file_size: max_file_size,
+          chunk_timeout: chunk_timeout,
+          chunk_timer: nil,
+          done?: false,
+          uploaded_size: 0
+        })
+
+      {:ok, socket}
+    else
+      {:error, :limit_exceeded} -> {:error, %{reason: :limit_exceeded}}
+      _ -> {:error, %{reason: :invalid_token}}
+    end
+  end
+
+  @impl true
+  def handle_in("chunk", {:binary, payload}, socket) do
```

Comment:
There's probably more work and that's probably why, but when testing uploads on an empty new project i get pattern match fail calling `handle_in("chunk", %{}, socket)`. Unsure that's supposed to happen, but thought i should report it :)